xml/System.IO/FileSystemWatcher.xml

```diff
@@ -823,20 +823,46 @@
+ you can use the following shell function to fix the problem:
+
+```sh
+handle_mui_error() {
+MUI_PATH=/proc/sys/fs/inotify/max_user_instances
+let mui=$(<"${MUI_PATH}")
+[[
+ "${1}" = "The configured user limit (${mui}) on the number of inotify instances has been reached"
+ ]] && let 'mui <<= 01' &&
+ echo $mui > "${MUI_PATH}" ||
+ pkexec sh -c ">'${MUI_PATH}' echo $mui"
+}
+# Example usage:
+handle_mui_error 'The configured user limit (128) on the number of inotify instances has been reached'
+```
+
+ Each execution of this script doubles the system-wide limit, except when run in parallel.
+ This script requires privilege elevation, so either run it as superuser
+ or be prepared to authenticate using [pkexec]((https://www.freedesktop.org/software/polkit).
+ This is a global setting that can be modified by another user at any time.
+ Use the configuration tools provided by your operating system to change the initial value of this setting at boot time.
+ This script does not attempt to fix the other possible reason for this exception,
+ which is *the per-process limit on the number of open file descriptors has been reached*
+ (see your operating system documentation for error `EMFILE`).
+
 ## Examples  
  The following example creates a <xref:System.IO.FileSystemWatcher> to watch the directory specified at run time. The component is set to watch for changes in `LastWrite` and `LastAccess` time, the creation, deletion, or renaming of text files in the directory. If a file is changed, created, or deleted, the path to the file prints to the console. When a file is renamed, the old and new paths print to the console.  
   
  :::code language="cpp" source="~/samples/snippets/cpp/VS_Snippets_CLR_Classic/classic NotifyFilters Example/CPP/source.cpp" id="Snippet1":::
  :::code language="csharp" source="~/samples/snippets/csharp/System.IO/FileSystemWatcher/Overview/source.cs" id="Snippet1":::
- :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic NotifyFilters Example/VB/source.vb" id="Snippet1":::  
-  
+ :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic NotifyFilters Example/VB/source.vb" id="Snippet1":::
+
  ]]></format>
         </remarks>
         <exception cref="T:System.ObjectDisposedException">The <see cref="T:System.IO.FileSystemWatcher" /> object has been disposed.</exception>
         <exception cref="T:System.PlatformNotSupportedException">The current operating system is not Microsoft Windows NT or later.</exception>
         <exception cref="T:System.IO.FileNotFoundException">The directory specified in <see cref="P:System.IO.FileSystemWatcher.Path" /> could not be found.</exception>
+        <exception cref="T:System.IO.IOException">The configured user limit on the number of <c>inotify</c> instances has been reached, or the per-process limit on the number of open file descriptors has been reached.</exception>
```

Comment:
I don't know of any "rules" but we typically do it [like this](https://github.com/dotnet/dotnet-api-docs/blob/bf815f4e86a4b5311e2ee1e5453226feef139c38/xml/System.IO/Directory.xml#L237). So in this case it would be:

```suggestion
        <exception cref="T:System.IO.IOException">Unix-based operating systems only: The configured user limit on the number of <c>inotify</c> instances has been reached, or the per-process limit on the number of open file descriptors has been reached.</exception>
```